astropy/units/core.py

```diff
@@ -1755,19 +1760,20 @@ def __call__(self, s, represents=None, format=None, namespace=None,
             if is_effectively_unity(represents.value):
                 represents = represents.unit
             else:
-                # cannot use _error_check=False: scale may be effectively unity
                 represents = CompositeUnit(represents.value *
                                            represents.unit.scale,
                                            bases=represents.unit.bases,
-                                           powers=represents.unit.powers)
+                                           powers=represents.unit.powers,
+                                           _error_check=False)
```

Comment:
Sorry for being a bit dense - I just want to make sure I follow the changes to `_error_check` and the new `_error_check=False`'s added.

In this case, why is it safe to not check?

Reply:
My logic was that here we know `represents` is a `Quantity`, and `Quantity` instances ensure their unit is properly initialized as a `Unit`. So, `represents.unit` is a `Unit` instance which needs no further checking (note that it has to have `scale`, `bases`, and `powers` attributes). Now possibly this could still go wrong. Since this is not on the fast path for anything anyway, shall I just change it back?